### PR TITLE
LEDs subsytem

### LINKED ISSUE #34: LED subsystem
Most important subsystem

### PR DESCRIPTION
**Pull request by owenflatman**
Created LEDs without IO. Then made commands
close #34

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/LEDs.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    private int length = Constants.LEDsConstants.LED_COUNT;
+    private int port = Constants.LEDsConstants.PWM_PORT;
+
+    /**
+     * constructs a LED Subsystem
+     *
+     * @param length length of the addressable LEDS
+     * @param port port ID for PWM
+     */
+    public LEDs(int length, int port) {
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/LEDs.java:24]: I don't need the arguments

File: src/main/java/frc/robot/subsystems/LEDs.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        controLedBuffer = new AddressableLEDBuffer(length);
+        addressableLED = new AddressableLED(port);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/LEDs.java:26]: This can be outside the constructor where it's defined.

**Review by legoguy1000 — CHANGES_REQUESTED** [review]

**Review by wilsonwatson — APPROVED** [review]
Looks like a kitchen sink, but otherwise is fine.

File: src/main/java/frc/robot/subsystems/climber/Climber.java [hunk 3]
```diff
@@ -36,6 +36,14 @@ public Climber(ClimberIO io) {
...
         io.updateInputs(inputs);
         Logger.processInputs("Climber", inputs);
+        double leftClimberPIDValue =
+            leftClimberPIDController.calculate(leftClimberDistanceTraveled());
+        double rightClimberPIDValue =
+            rightClimberPIDController.calculate(rightClimberDistanceTraveled());
+        double climberFeedForwardValue = climberFeedforward.calculate(0, 0);
+        io.setLeftClimberVoltage(climberFeedForwardValue + leftClimberPIDValue);
+        io.setRightClimberVoltage(climberFeedForwardValue + rightClimberPIDValue);
+        Logger.recordOutput("/Climber/Feedforward", climberFeedForwardValue);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/climber/Climber.java:46]: IDK why this is here but shouldn't be changing things in CLimber in this PR.  probably a merge issue.

File: src/main/java/frc/robot/subsystems/elevator_wrist/ElevatorWrist.java [hunk 4]
```diff
@@ -61,11 +61,9 @@ public void periodic() {
         double elevatorPIDValue = elevatorPIDController.calculate(elevatorDistanceTraveled());
         double wristPIDValue = wristPIDController.calculate(inputs.wristAbsoluteEncRawValue);
 
-        double elevatorFeedForwardValue =
-            elevatorFeedForward.calculate(0, 0, elevatorPIDController.getPeriod());
+        double elevatorFeedForwardValue = elevatorFeedForward.calculate(0, 0);
 
-        double wristFeedForwardValue =
-            wristFeedForward.calculate(0, 0, wristPIDController.getPeriod());
+        double wristFeedForwardValue = wristFeedForward.calculate(0, 0);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/elevator_wrist/ElevatorWrist.java:66]: IDK why this is here but shouldn't be changing things in Elevator in this PR.  probably a merge issue.

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -343,18 +351,18 @@ public static final class PID {
...
             public static final double WRIST_KI = 0;
             public static final double WRIST_KD = 0;
             public static final double WRIST_MAX_VELOCITY = 0;
             public static final double WRIST_MAX_ACCELERATION = 0;
-            public static final double WRIST_KS = 0;
-            public static final double WRIST_KG = 0;
-            public static final double WRIST_KV = 0;
+            public static final double WRIST_KS = 0.001;
+            public static final double WRIST_KG = 0.001;
+            public static final double WRIST_KV = 0.001;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/Constants.java:365]: Same as other comments

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -45,6 +50,9 @@ public class RobotContainer {
     private Swerve s_Swerve;
     private Shooter shooter;
     private Intake intake;
+    private ElevatorWrist elevatorWrist;
+    private Climber climber;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:54]: same

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -17,7 +18,11 @@
 import frc.lib.util.photon.PhotonReal;
 import frc.robot.Robot.RobotRunType;
 import frc.robot.commands.CommandFactory;
-import frc.robot.commands.TeleopSwerve;
+import frc.robot.commands.leds.MovingColorLEDs;
+import frc.robot.commands.swerve.TeleopSwerve;
+import frc.robot.subsystems.LEDs;
+import frc.robot.subsystems.climber.Climber;
+import frc.robot.subsystems.elevator_wrist.ElevatorWrist;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:25]: same

**Review by legoguy1000 — CHANGES_REQUESTED** [review]
Lets take a few min and figure out what LED functions/actions we actually want and then only merge in what we need. Not everything. If we're not going to use it, lets keep it out of the code for now since it just clutters things.

**Comment by legoguy1000** [comment]
lets add a trigger to robot container that flashes the LEDs whenever we have a note

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -130,11 +137,14 @@ private void configureButtonBindings() {
...
         driver.b().whileTrue(intake.runIndexerMotor(-.1));
         driver.x().whileTrue(CommandFactory.passThroughShoot(shooter, intake));
-
         operator.x().whileTrue(CommandFactory.spit(shooter, intake));
         operator.rightTrigger().whileTrue(CommandFactory.shootSpeaker(shooter, intake));
         operator.start().onTrue(elevatorWrist.ampPosition());
         operator.back().onTrue(elevatorWrist.homePosition());
+        operator.povUp().onTrue(new FlashingLEDColor(leds, Color.kOrange).withTimeout(5));
+        operator.povDown().onTrue(new FlashingLEDColor(leds, Color.kBlue).withTimeout(5));
+        new Trigger(() -> intake.getSensorStatus())
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:146]: Here's what we did last year. https://github.com/Frc5572/FRC2023/blob/b01038bf4f19c6f16ed4cc5fe20e5728422f1238/src/main/java/frc/robot/RobotContainer.java#L203

Comment (owenflatman) [reply src/main/java/frc/robot/RobotContainer.java:146]: What do you mean? Why does the current code not work?

Comment (owenflatman) [reply src/main/java/frc/robot/RobotContainer.java:146]: Nvm I understand

**Comment by wilsonwatson** [comment]
This PR has become stale. Minimal LEDs are included in #82.

**Pull request closed without merging**